Editor/Manifest/Version.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace StansAssets.Foundation.Editor
+{
+    public class Version
+    {
+        // Regular Expression to check a Semantic Version string. With numbered capture groups
+        // (so cg1 = major, cg2 = minor, cg3 = patch, cg4 = prerelease and cg5 = buildmetadata)
+        // that is compatible with ECMA Script (JavaScript),
+        // PCRE (Perl Compatible Regular Expressions, i.e. Perl, PHP and R), Python and Go.
+        // See: https://regex101.com/r/vkijKf/1/
+        static readonly Regex s_Matcher = new Regex(@"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$");
+
+        public readonly int Major;
+        public readonly int Minor;
+        public readonly int Patch;
+
+        public readonly bool HasPreRelease;
+        public readonly string PreRelease;
+
+        public readonly bool HasBuildMetadata;
+        public readonly string BuildMetadata;
```

Comment:
I agree with Codacy here, let's use properties installed if fields for the public API.
Found an interesting point about this on stack overflow:


See this article http://blog.codinghorror.com/properties-vs-public-variables/

Specifically

Reflection works differently on variables vs. properties, so if you rely on reflection, it's easier to use all properties.
You can't databind against a variable.
Changing a variable to a property is a breaking change.